bootstrap/bootstrapper/fs/source.go

```diff
@@ -88,18 +71,25 @@ func (fss *fileSystemSource) GetAvailability(shard uint32, targetRangesForShard
 
 // ReadData returns raw series for a given shard within certain time ranges.
 func (fss *fileSystemSource) ReadData(shard uint32, tr xtime.Ranges) (m3db.ShardResult, xtime.Ranges) {
-	log := fss.opts.GetLogger()
-
 	if xtime.IsEmpty(tr) {
 		return nil, tr
 	}
-	files, err := fss.getInfoFiles(shard)
-	if err != nil {
-		log.Errorf("unable to get info files for shard %d: %v", shard, err)
+
+	var files []string
+	fs.ForEachInfoFile(
+		fss.filePathPrefix,
+		shard,
+		func(fname string, _ []byte) {
+			files = append(files, fname)
+		},
+	)
```

Comment:
This is more stylistic so take or leave it, and this is more just leaving a comment for comment sake.  I like to write these like this:
```go
fs.ForEachInfoFile(fss.filePathPrefix, shard, func(fname string, _ []byte) {
  files = append(files, fname)
})
```